azure-kusto-data/source/client.ts

```diff
@@ -39,11 +41,20 @@ export class KustoClient {
             [ExecutionType.QueryV1]: `${this.cluster}/v1/rest/query`,
         };
         this.aadHelper = new AadHelper(this.connectionString);
-        this.headers = {
+        const headers = {
             "Accept": "application/json",
             "Accept-Encoding": "gzip,deflate",
             "x-ms-client-version": `Kusto.Node.Client:${pkg.version}`,
+            "Connection": "Keep-Alive",
         };
+        this.axiosInstance = axios.create({
+            headers,
+            validateStatus: (status: number) => status == 200,
```

Comment:
Since the PR is about making sure we throw on non 200, we should probably have a test that makes sure that an exception is thrown.